packages/selection/src/useSelection.js

```diff
@@ -0,0 +1,270 @@
+      }
+
+      // eslint-disable-next-line no-case-declarations
+      const updatedState = { ...state };
+
+      if (!isSelectControlled) {
+        updatedState.selectedItem = action.payload;
+      }
+
+      if (!isFocusControlled) {
+        updatedState.focusedItem = undefined;
+      }
+
+      return updatedState;
+    case ACTIONS.KEYBOARD_SELECT:
+      if (onSelect) {
+        onSelect(action.payload);
+
+        return state;
+      }
+
+      return { ...state, selectedItem: action.payload };
+    case ACTIONS.EXIT_WIDGET:
+      if (onFocus) {
+        onFocus(undefined);
+
+        return state;
+      }
+
+      return { ...state, focusedItem: undefined };
+    default:
+      return new Error('Invalid reducer action');
+  }
+}
+/* eslint-enable indent */
+
+/**
+ * Custom hook to manage selection using the Roving Tab Index strategy
+ *
+ * https://www.w3.org/TR/wai-aria-practices/#kbd_roving_tabindex
+ */
+export function useSelection({
+  direction = DIRECTIONS.HORIZONTAL,
+  defaultFocusedIndex = 0,
+  selectedItem,
+  focusedItem,
+  onSelect,
+  onFocus
+} = {}) {
+  const { rtl } = useContext(LocaleContext);
+
+  const refs = [];
+  const items = [];
+
+  const [state, dispatch] = useReducer(
+    (reducerState, action) =>
+      stateReducer(reducerState, action, { onSelect, onFocus, selectedItem, focusedItem }),
+    { selectedItem, focusedItem }
+  );
+
+  const controlledFocusedItem = getControlledValue(focusedItem, state.focusedItem);
+  const controlledSelectedItem = getControlledValue(selectedItem, state.selectedItem);
+
+  useEffect(
+    () => {
+      if (controlledFocusedItem !== undefined) {
+        const focusedIndex = items.indexOf(controlledFocusedItem);
+
+        refs[focusedIndex] && refs[focusedIndex].current.focus();
+      }
+    },
+    [controlledFocusedItem]
+  );
+
+  const getContainerProps = ({ role = 'listbox', ...other } = {}) => ({
+    role,
+    ...other
+  });
+
+  const getItemProps = ({
+    selectedAriaKey = 'aria-selected',
+    role = 'option',
+    onFocus: onFocusCallback,
+    onKeyDown,
+    onClick,
+    item,
+    focusRef,
+    ...other
+  } = {}) => {
+    if (item === undefined) {
+      throw new Error('Accessibility Error: You must provide an "item" option to "getItemProps()"');
+    }
+
+    if (focusRef === undefined) {
+      throw new Error(
+        'Accessibility Error: You must provide a "focusRef" option to "getItemProps()"'
+      );
+    }
+
+    refs.push(focusRef);
+    items.push(item);
+
+    const isSelected = controlledSelectedItem === item;
+    const isFocused =
+      controlledFocusedItem === undefined ? isSelected : controlledFocusedItem === item;
+    const tabIndex =
+      isFocused ||
+      (state.selectedItem === undefined &&
+        state.focusedItem === undefined &&
+        items.indexOf(item) === defaultFocusedIndex)
+        ? 0
+        : -1;
+    const reDirVertical = new RegExp(`^(${DIRECTIONS.VERTICAL}|${DIRECTIONS.BOTH})$`, 'u');
```

Comment:
Even though a RegEx technical works here, would it be prudent to treat this as an ENUM?

The fact that it's a string is an implementation detail, but I find it much more readable to not test regex, just have a comparison:

```javascript
const isDirVertical = direction === DIRECTIONS.VERTICAL || direction === DIRECTIONS.BOTH;
```